iris_grib/_save_rules.py

```diff
@@ -693,22 +694,45 @@ def grid_definition_section(cube, grib):
 ###############################################################################
 
 def set_discipline_and_parameter(cube, grib):
-    # NOTE: for now, can match by *either* standard_name or long_name.
-    # This allows workarounds for data with no identified standard_name.
-    grib2_info = gptx.cf_phenom_to_grib2_info(cube.standard_name,
-                                              cube.long_name)
-    if grib2_info is not None:
-        gribapi.grib_set(grib, "discipline", grib2_info.discipline)
-        gribapi.grib_set(grib, "parameterCategory", grib2_info.category)
-        gribapi.grib_set(grib, "parameterNumber", grib2_info.number)
-    else:
-        gribapi.grib_set(grib, "discipline", 255)
-        gribapi.grib_set(grib, "parameterCategory", 255)
-        gribapi.grib_set(grib, "parameterNumber", 255)
+    # Default values for parameter identity keys = effectively "MISSING".
+    discipline, category, number = 255, 255, 255
+    identity_found = False
+
+    # First, see if we can find and interpret a 'GRIB_CODING' attribute.
```

Comment:
I'm not convinced its a good idea to interpret the "GRIB_CODING" attribute first.
I think the standard_name/long_name should take precedence when determining the correct discipline/category/number as that it is the primary means we have of identifying a cube. This reliance on GRIB_CODING means in future we have to take extra care with attributes and I don't want that requirement. 
For example someone take a copy of a cube, update the data array, and only update the name, then save it out as a cube

Reply:
From an offline discussion with @pp-mo, it turns out that this is how the STASH attribute is handled :warning: :warning: :warning: 

In which case, my preference is for matching that behaviour so I'm happy with this as it is.